src/discussions/comments/data/slices.js

```diff
@@ -16,26 +16,32 @@ const commentsSlice = createSlice({
     commentsById: {
       // Map comment ids to comments.
     },
-    pages: [],
     // Stores the comment being posted in case it needs to be reposted due to network failure.
     // TODO: save in localstorage so user can continue editing?
     commentDraft: null,
-    totalPages: null,
-    totalThreads: null,
     postStatus: RequestStatus.SUCCESSFUL,
+    pagination: {
+    },
   },
   reducers: {
     fetchCommentsRequest: (state) => {
       state.status = RequestStatus.IN_PROGRESS;
     },
     fetchCommentsSuccess: (state, { payload }) => {
       state.status = RequestStatus.SUCCESSFUL;
-      state.pages[payload.page - 1] = payload.ids;
-      state.commentsInThreads = { ...state.commentsInThreads, ...payload.commentsInThreads };
+      Object.keys(payload.commentsInThreads).forEach(threadId => {
+        state.commentsInThreads[threadId] = [
+          ...state.commentsInThreads[threadId] || [],
```

Comment:
I think to make this clearer let's rewrite this as:
```suggestion
          ...(state.commentsInThreads[threadId] || []),
```